roundabout-core/src/test/java/at/fhv/itm3/s2/roundabout/integration/DestinationReachedIntegration.java

```diff
@@ -0,0 +1,54 @@
+package at.fhv.itm3.s2.roundabout.integration;
+
+import at.fhv.itm3.s2.roundabout.RoundaboutSimulationModel;
+import at.fhv.itm3.s2.roundabout.api.entity.AbstractSource;
+import at.fhv.itm3.s2.roundabout.api.entity.ICar;
+import at.fhv.itm3.s2.roundabout.api.entity.IRoute;
+import at.fhv.itm3.s2.roundabout.api.entity.Street;
+import at.fhv.itm3.s2.roundabout.mocks.RouteGenerator;
+import at.fhv.itm3.s2.roundabout.mocks.RouteType;
+import desmoj.core.simulator.Experiment;
+import desmoj.core.simulator.TimeInstant;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.mockito.Mockito.when;
+
+public class DestinationReachedIntegration {
+
+    private RoundaboutSimulationModel model;
+    private Experiment exp;
+
+    @Before
+    public void setUp() {
+        model = new RoundaboutSimulationModel(null, "", false, false);
+        exp = new Experiment("RoundaboutSimulationModel Experiment");
+        model.connectToExperiment(exp);
+        exp.setShowProgressBar(false);
+    }
+
+    @Test
+    public void destinationReached() {
+        exp.stop(new TimeInstant(60, TimeUnit.SECONDS));
+
+        RouteGenerator routeGenerator = new RouteGenerator(model);
+        IRoute route = routeGenerator.getRoute(RouteType.TWO_STREETSECTIONS_ONE_CAR);
+        AbstractSource source = route.getSource();
+        Street destinationSection = route.getDestinationSection();
+
+        source.startGeneratingCars();
+
+        exp.start();
+        exp.finish();
+
+        if(!destinationSection.isEmpty()){
+            Assert.assertEquals("car never reached destination.",31, destinationSection.getIdentNumber());
```

Comment:
Ich verstehe nicht ganz, was du mit diesem Test meinst. Ich hätte mir erwartet, dass überprüft wird, ob das Auto die Destination erreicht hat, aber für mich sieht es so aus, als würdest du nur überprüfen, ob die DestinationSection die Nummer 31 hat. Dadurch ist doch nicht überprüft, ob das Auto die Destination erreicht hat, oder? Ich würde überprüfen, ob in der DestinationSection ein Car ist. Sonst sehe ich den Sinn dahinter nicht.